### PR TITLE
reconcile service resource

### PR DESCRIPTION
**Pull request by calvix**
As the annotation is now needed for guest cluster monitoring we need to be able to update old cluster and the that means enable reconciliation of the service to desired state

Also since the etcd port is different across providers and it needs to be part of annotation

### CODE REVIEW

File: service/controller/v13/key/key.go [hunk 1]
```diff
@@ -168,7 +168,7 @@ func ClusterCustomer(customObject v1alpha1.AWSConfig) string {
 }
 
 func ClusterEtcdDomain(customObject v1alpha1.AWSConfig) string {
-	return customObject.Spec.Cluster.Etcd.Domain
+	return fmt.Sprintf("%s:%d", customObject.Spec.Cluster.Etcd.Domain, customObject.Spec.Cluster.Etcd.Port)
```
Comment (calvix) [inline service/controller/v13/key/key.go]: I had to fix this bug as part of my changes were already in v13 but the rest was not and I didn't want to block the release. Adding etcd port as we have different ports across providers

**Comment by xh3b4sd** [comment]
So this looks like we have to refactor it somehow. The code seems to be basically the same between the providers. Would it make sense to extract the service resource to its own repo and have it implemented and tested once? Like what is the common ground here or where does it differ across providers? All the code duplication cannot be right.

**Comment by calvix** [comment]
> So this looks like we have to refactor it somehow. The code seems to be basically the same between the providers. Would it make sense to extract the service resource to its own repo and have it implemented and tested once? Like what is the common ground here or where does it differ across providers? All the code duplication cannot be right.

there is a difference as `kvm-operator` works with `[]v1.Service` (as it has master and worker service) and other operators have only `v1.Service` (as they have only master service) 

I am not sure about the library, as it feels really weird that if I want to change something in the resource I should go to the separate repository, AFAIK we keep all resources that operator manages in same repo.

**Comment by xh3b4sd** [comment]
We already have and are going to use https://github.com/giantswarm/statusresource even more. The common ground is the slice of services so this works for all providers. What is different can be configured. Easy.

**Comment by kopiczko** [comment]
For service it would make sense to have it in the operatorkit.

**Comment by xh3b4sd** [comment]
Not sure because it is quite Giant Swarm specific and opinionated, isn't it? Why should somebody reuse it?

**Comment by calvix** [comment]
this is for sure really custom service so it doesn't make sense to have it in operatorkit

**Comment by xh3b4sd** [comment]
I would vote for separate repos for each resource because of the dependency hell we are going to get otherwise.

**Comment by calvix** [comment]
RFR

**Pull request merged**